go/host/host.go

```diff
@@ -393,11 +376,17 @@ func (h *host) waitForEnclave() {
 }
 
 // starts the host main processing loop
-// todo: matt to fix this complexity `nolint` in next PR with block provider
-func (h *host) startProcessing() { //nolint:gocognit
+func (h *host) startProcessing() {
 	// Only open the p2p connection when the host is fully initialised
```

Comment:
@tudor-malene Yeah interesting, I guess so....

Presumably as an optimisation rather than for correctness? (I.e. don't want to bother a host with p2p transactions etc. that it's not going to do anything with while it's streaming through history).

Maybe I'll put a todo in the p2p consuming part to check if we should be discarding updates when we're in certain states? Or better to have an "initial sync" flag that starts true and gets set to false the first time we process a live block? It feels inconsistent that we do that for initial load but not if the host falls behind for some reason.

Reply:
Pls raise a ticket to think about this at some point. We can have a look at geth